src/tests/DbgShim.UnitTests/DbgShimTests.cs

```diff
@@ -0,0 +1,556 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.DebugServices;
+using Microsoft.Diagnostics.Runtime.Utilities;
+using Microsoft.Diagnostics.TestHelpers;
+using SOS.Hosting;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Extensions;
+
+namespace Microsoft.Diagnostics
+{
+    public class DbgShimTests : IDisposable
+    {
+        private const string ListenerName = "DbgShimTests";
+
+        public static IEnumerable<object[]> GetConfigurations(string key, string value)
+        {
+            return TestRunConfiguration.Instance.Configurations.Where((c) => key == null || c.AllSettings.GetValueOrDefault(key) == value).Select(c => new[] { c });
+        }
+
+        public static IEnumerable<object[]> Configurations => GetConfigurations("TestName", null);
+
+        private ITestOutputHelper Output { get; }
+
+        public DbgShimTests(ITestOutputHelper output)
+        {
+            Output = output;
+            LoggingListener.EnableListener(output, ListenerName);
+        }
+
+        void IDisposable.Dispose() => Trace.Listeners.Remove(ListenerName);
+
+        /// <summary>
+        /// Test RegisterForRuntimeStartup for launch
+        /// </summary>
+        [SkippableTheory, MemberData(nameof(Configurations))]
+        public async Task Launch1(TestConfiguration config)
+        {
+            await RemoteInvoke(config, nameof(Launch1), static async (string xml) =>
```

Comment:
```suggestion
            await RemoteInvoke(config, nameof(Launch1), static async (string configXml) =>
```
I couldn't initially figure out what this 'xml' variable was. How about we adjust the name?

Reply:
Sounds good.